tests/syntax/test_errors.py

```diff
@@ -79,52 +56,52 @@ def test_new_python_class():
 ## Soft requirements
 
 def test_malformed_soft_requirement():
-    with pytest.raises(TokenParseError):
+    with pytest.raises(SyntaxError):
         compileScenic('require[x] 3 == 3')
-    with pytest.raises(TokenParseError):
+    with pytest.raises(SyntaxError):
         compileScenic('require[1+x] 3 == 3')
-    with pytest.raises(TokenParseError):
+    with pytest.raises(SyntaxError):
         compileScenic('require[] 3 == 3')
 
 ## Specifiers
 
 def test_undefined_specifier():
-    with pytest.raises(TokenParseError):
+    with pytest.raises(SyntaxError):
         compileScenic('Object cattywampus')
-    with pytest.raises(TokenParseError):
+    with pytest.raises(SyntaxError):
         compileScenic('Object athwart 3')
 
 ## Illegal usages of keywords
 
 def test_reserved_functions():
-    with pytest.raises(TokenParseError):
+    with pytest.raises(SyntaxError):
         compileScenic('PropertyDefault()')
 
 ## Unmatched parentheses and multiline strings
 
 def test_unmatched_parentheses():
-    with pytest.raises(TokenParseError):
+    with pytest.raises(TokenError):
```

Comment:
The tokenizer raises `TokenError` for errors like unmatched parenthesis. CPython uses the internal state of the tokenizer to produce more readable error messages, but it seems that the state is not accessible from Python. Need more investigation on this matter.

Reply:
Interesting. Given that we didn't provide more helpful errors with the old parser I think we can leave this for later (I see you made a TODO item).